plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenPath.java

```diff
@@ -0,0 +1,374 @@
+      return ext;
+    }
+  }
+
+  public static class Coordinates
+  {
+    private final boolean snapshot;
+
+    private final String groupId;
+
+    private final String artifactId;
+
+    private final String version;
+
+    private final String baseVersion;
+
+    private final String classifier;
+
+    private final String extension;
+
+    private final SignatureType signatureType;
+
+    public Coordinates(final boolean snapshot,
+                       final String groupId,
+                       final String artifactId,
+                       final String version,
+                       final String baseVersion,
+                       final String classifier,
+                       final String extension,
+                       final SignatureType signatureType)
+    {
+      this.snapshot = snapshot;
+      this.groupId = checkNotNull(groupId);
+      this.artifactId = checkNotNull(artifactId);
+      this.version = checkNotNull(version);
+      this.baseVersion = checkNotNull(baseVersion);
+      this.classifier = classifier;
+      this.extension = checkNotNull(extension);
+      this.signatureType = signatureType;
+    }
+
+    public boolean isSnapshot() {
+      return snapshot;
+    }
+
+    @Nonnull
+    public String getGroupId() {
+      return groupId;
+    }
+
+    @Nonnull
+    public String getArtifactId() {
+      return artifactId;
+    }
+
+    @Nonnull
+    public String getVersion() {
+      return version;
+    }
+
+    @Nonnull
+    public String getBaseVersion() {
+      return baseVersion;
+    }
+
+    @Nullable
+    public String getClassifier() {
+      return classifier;
+    }
+
+    @Nonnull
+    public String getExtension() {
+      return extension;
+    }
+
+    @Nullable
+    public SignatureType getSignatureType() {
+      return signatureType;
+    }
+  }
+
+  private final Kind kind;
+
+  private final String path;
+
+  private final String fileName;
+
+  private final HashType hashType;
+
+  private final Coordinates coordinates;
+
+  public MavenPath(final String path, final Coordinates coordinates)
+  {
+    this.path = checkNotNull(path);
+    this.fileName = this.path.substring(path.lastIndexOf('/') + 1);
+    HashType ht = null;
+    for (HashType v : HashType.values()) {
+      if (this.fileName.endsWith("." + v.getExt())) {
+        ht = v;
+        break;
+      }
+    }
+    this.hashType = ht;
+    this.coordinates = coordinates;
+    if (coordinates != null) {
+      this.kind = Kind.ARTIFACT;
+    }
+    else if (main().getFileName().equals(MAVEN_METADATA_FILENAME)) {
+      this.kind = Kind.METADATA;
+    }
+    else {
+      this.kind = Kind.OTHER;
+    }
+  }
+
+  @Nonnull
+  public Kind getKind() {
+    return kind;
+  }
+
+  @Nonnull
+  public String getPath() {
+    return path;
+  }
+
+  @Nonnull
+  public String getFileName() {
+    return fileName;
+  }
+
+  @Nullable
+  public HashType getHashType() {
+    return hashType;
+  }
+
+  @Nullable
```

Comment:
Bears repeating in javadoc: artifact paths (or their hashes and signatures) have coordinates, but mere file-paths don't. This is used to bifurcate a lot of behavior in the maven facet. An isArtifactPath() method would be good, too.